addon/-private/utils/headers.js

```diff
@@ -0,0 +1,82 @@
+/* global Symbol */
+// Partially implements Headers from the Fetch API
+// https://developer.mozilla.org/en-US/docs/Web/API/Headers
+function HeadersPolyfill(headers) {
+  headers = headers || {};
+  this.headers = {};
+
+  for (var header in headers) {
+    this.headers[header] = headers[header].split(', ');
+  }
+}
+
+HeadersPolyfill.prototype.append = function(header, value) {
+  header = header.toLowerCase();
+  if (!this.has(header)) {
+    this.headers[header] = [];
+  }
+
+  this.headers[header].push(value);
+};
+
+HeadersPolyfill.prototype.delete = function(header) {
+  delete this.headers[header.toLowerCase()];
+};
+
+HeadersPolyfill.prototype.entries = function() {
+  var entries = [];
+
+  for (var key in this.headers) {
+    var values = this.headers[key];
+    for (var index = 0; index < values.length; ++index ) {
+      entries.push([key, values[index]]);
+    }
+  }
+
+  return entries[Symbol.iterator]();
+};
+
+HeadersPolyfill.prototype.get = function(header) {
```

Comment:
Since there is a `get` method this may cause a user to assume this is an Ember Object.